editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java

```diff
@@ -47,18 +48,18 @@ public boolean supportsDSL(CamelDSLType type) {
 	 * @return	true on success, otherwise false
 	 * @throws CoreException	on any error
 	 */
-	public final boolean create(IProject project, NewProjectMetaData projectMetaData) throws CoreException {
-		IProgressMonitor monitor = new NullProgressMonitor();
+	public final boolean create(IProject project, NewProjectMetaData projectMetaData, IProgressMonitor monitor) throws CoreException {
+		SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
```

Comment:
The division of work should reflect the proportion.
In other places, there are often long-running tasks for which I wasn't able/don't take time to guess/determine the proportions, so I just put '1' for each of them.
I tried to be more precise at this place